### PR TITLE
PR: Overhaul "aces-dev" Reference Config Generator

### PR DESCRIPTION
**Pull request by KelSolaar**
Hi,

Allright, so this is a big one!

I first overhauled the [spreadsheet](https://docs.google.com/spreadsheets/d/1SXPt-USy3HlV2G2qAvh9zit6ZCINDOlfKT07yXJdWLg) so that we can generate a pivot table that can then be exported as a [CSV file](https://github.com/AcademySoftwareFoundation/OpenColorIO-Config-ACES/blob/d730b2daf2204fa7068ed0b68484848e104cb5be/opencolorio_config_aces/config/reference/generate/resources/OpenColorIO-ACES-Config%20Transforms%20-%20Reference%20Config%20-%20Mapping.csv).

This file basically maps the *aces-dev* CTL Transforms to the *OpenColorIO* BuiltinTransform"s". The discovery code, not the graph one, is used to check that the CSV file CTL Transforms exist and another check verifies that the BuiltinTransform"s" exist, from that point a few hops to generate the `ConfigData`, which is then passed to the almost OpenColorIO v2 feature complete `opencolorio_config_aces.generate_config` definition and we get close to our final "aces-dev" Reference config.

Various things in random order:

1. This is not final and I have not tested the config at all and I don't think it works!
2. We really need the ACEStransformID to be fixed.
3. Given that we do not have any IDTs mapped beside the Venice, I think we should make a PR to get CSC variants of them and call it a day, i.e. no IDTs.
4. We should PR descriptions to the CSC and maybe standardise a mini one-liner for each one.
5. I need to add support to clamp, errrr range, the ViewTransform"s", damn I don't like that RangeTranform name...
6. It is pretty much impossible to track which CTL Transform correspond to which ViewTransform from the config, so I will add them to the description.
7. Some inconsistency in the builtins names: REC709 vs REC.709
8. The graph code is optional now, I keep it around because it has been and is still extremely useful.
9. I need to pop the sRGB display to make it first.
5. The non Gitflow workflow is still a pain as you can see in the commits list.

Reference Config: https://gist.github.com/KelSolaar/1e2b054080e074e7f3529439470ee643/802a3cdf4edb203095f93e6c2e9f94379b9ce505

### CODE REVIEW

**Comment by KelSolaar** [comment]
I implemented support for Scott fork with the patched ACEStransformID: https://github.com/scottdyer/aces-dev/commit/4b88ef35afc41e58ea52d9acde68af24e75b58c5 and improved the description management.

**Review by michdolan — APPROVED** [review]
Two small comments. Looks great! Thanks @KelSolaar !

**Comment by chrisbrejon** [comment]
Hello guys,

just wanted to say we tried to load the config in our DCC software and we maybe noticed that the definition of the XYZ colorspace was missing ? Is the White Point for XYZ, D65 ?

I was only wondering about the roles. Do we want to keep them that way or would we rather have a couple more like color_picking, rendering or texture_paint ?

Thanks,
Chris

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (michdolan) [reply opencolorio_config_aces/config/reference/generate/config.py:839]: @doug-walker for the `ROLE_INTERCHANGE_DISPLAY` definition, is it expected to be a scene reference color space, display reference color space, or does it not matter?

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (doug-walker) [inline opencolorio_config_aces/config/reference/generate/config.py:631]: This would be a discussion for the group, but we may want to add "Display - " in front of these names.

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (doug-walker) [reply opencolorio_config_aces/config/reference/generate/config.py:773]: Also, you may want to make it an inactive color space, since it is more for use by OCIO rather than by artists.

**Review by doug-walker — CHANGES_REQUESTED** [review]
Nice work Thomas!  

It would be nice to use some of the other v2 features such as categories and viewing rules, but that could be done in a follow-up.

**Comment by doug-walker** [comment]
> just wanted to say we tried to load the config in our DCC software and we maybe noticed that the definition of the XYZ colorspace was missing ? Is the White Point for XYZ, D65 ?
> 
> I was only wondering about the roles. Do we want to keep them that way or would we rather have a couple more like color_picking, rendering or texture_paint ?

@chrisbrejon , the CIE-XYZ-D65 space should probably not be visible (I suggested Thomas add it to the inactive list as it is not useful for artists).  Regarding the white point, CIE XYZ is not really a color space by itself and does not have a fixed white point.  We are adding some things such as D65 adaptive white on top of CIE-XYZ here to make it more of a color space.

Michael and I suggested adding some roles.  We may want to discuss color_picking in the group since I know you have raised pros and cons to different possibilities.

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (KelSolaar) [reply opencolorio_config_aces/config/reference/generate/config.py:824]: Yes!

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (KelSolaar) [reply opencolorio_config_aces/config/reference/generate/config.py:839]: Yes!

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (KelSolaar) [reply opencolorio_config_aces/config/reference/generate/config.py:852]: Categories will need to be defined in the spreadsheet. Vewing rules, I haven't looked at those yet at all but happy to do so!

WRT to the Raw as Shared View, I'm not sure how to to that, the Python API requires a View Transform but your proto does not have any which left me puzzled, had more important stuff to do, so I moved on :)

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (KelSolaar) [reply opencolorio_config_aces/config/reference/generate/config.py:589]: There are a few options with an enum to configure the final description, worth poking at!

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (KelSolaar) [reply opencolorio_config_aces/config/reference/generate/config.py:773]: Yes and yes! I originally had it as a default in an earlier toy implementation where I had a `display_factory` definition but I culled it and forgot to add that back.

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (KelSolaar) [reply opencolorio_config_aces/config/reference/generate/config.py:853]: The order is currently dictated by the spreadsheet so we should do that there if we need/want!

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (KelSolaar) [reply opencolorio_config_aces/config/reference/generate/config.py:439]: No strong opinion here but let's talk about it tomorrow!

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (KelSolaar) [reply opencolorio_config_aces/config/reference/generate/config.py:631]: Ditto!

**Review by michdolan — APPROVED** [review]

**Review by michdolan — CHANGES_REQUESTED** [review]
From topics discussed working group meeting:
- Set "Un-tone-mapped" as default view transform via Config API
- Change blue light fix to Look
- Add display prefix to family as well as name

I think that was all. Thanks again!

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (doug-walker) [reply opencolorio_config_aces/config/reference/generate/config.py]: Proposed description string: "This 'identity view' is useful for presenting the source colorimetry directly on the display without any tone-mapping. For example, it may be useful for viewing textures as they would appear in an image editing program. It is not an official ACES transform."

And as Michael mentioned in the meeting, please call setDefaultViewTransformName on the config object to set this as the default view transform.

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (doug-walker) [reply opencolorio_config_aces/config/reference/generate/config.py:858]: It works the same as the other shared views except use an empty string for the view transform.

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (doug-walker) [reply opencolorio_config_aces/config/reference/generate/config.py:859]: I copied the original set and pasted below and re-ordered them.  I also added proposed encoding and category values.  Once the group agrees, these could be pasted over the originals and a new csv file could be exported.

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (doug-walker) [reply opencolorio_config_aces/config/reference/generate/config.py:439]: After thinking about this some more, I still feel that "Input - Canon - CLog3 CGamut" is better than "ACES - CLog3 CGamut", but I acknowledge that not everyone in the group agreed during the meeting.  I'll propose a compromise: "ACES - CSC - CLog3 CGamut".  That way there would at least be something to indicate that there is a difference between the color spaces that are actual ACES standards (ACES2065-1, ACEScg, ACEScct, ACEScc) and the vendor spaces.  But I'm not opposed to putting out what we have now for comment.

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (KelSolaar) [reply opencolorio_config_aces/config/reference/generate/config.py:859]: Cannot see your ordered displays but I added an ordering column to the spreadsheet to use with the pivot table and it works, should be easy to tweak the order now!

```yaml
  sRGB:
    - !<Views> [ACES 1.0 SDR-video, ACES 1.0 SDR-video (D60 sim on D65), Un-tone-mapped, Log, Raw]
  Rec.1886 / Rec.709 video:
    - !<Views> [ACES 1.0 SDR-video, ACES 1.0 SDR-video (D60 sim on D65), Un-tone-mapped, Log, Raw]
  Rec.1886 / Rec.2020 video:
    - !<Views> [ACES 1.0 SDR-video, ACES 1.0 SDR-video (D60 sim on D65), ACES 1.1 SDR-video (Rec.709 lim), ACES 1.1 SDR-video (P3 lim), Un-tone-mapped, Log, Raw]
  P3-D60:
    - !<Views> [ACES 1.0 SDR-cinema, Un-tone-mapped, Log, Raw]
  P3-D65:
    - !<Views> [ACES 1.0 SDR-cinema, ACES 1.1 SDR-cinema (D60 sim on D65), ACES 1.1 SDR-cinema (Rec.709 lim), Un-tone-mapped, Log, Raw]
  P3-DCI:
    - !<View> {name: ACES 1.0 SDR-cinema, view_transform: ACES 1.0 SDR-cinema, display_colorspace: P3-DCI}
    - !<View> {name: ACES 1.0 SDR-cinema (D60 sim on DCI), view_transform: ACES 1.0 SDR-cinema (D60 sim on DCI), display_colorspace: P3-DCI}
    - !<View> {name: ACES 1.1 SDR-cinema (D65 sim on DCI), view_transform: ACES 1.1 SDR-cinema (D65 sim on DCI), display_colorspace: P3-DCI}
    - !<Views> [Un-tone-mapped, Log, Raw]
  ST2084-P3-D65:
    - !<View> {name: ACES 1.1 HDR-video (1000 nits), view_transform: ACES 1.1 HDR-video (1000 nits & P3 lim), display_colorspace: ST2084-P3-D65}
    - !<View> {name: ACES 1.1 HDR-video (2000 nits), view_transform: ACES 1.1 HDR-video (2000 nits & P3 lim), display_colorspace: ST2084-P3-D65}
    - !<View> {name: ACES 1.1 HDR-video (4000 nits), view_transform: ACES 1.1 HDR-video (4000 nits & P3 lim), display_colorspace: ST2084-P3-D65}
    - !<View> {name: ACES 1.1 HDR-cinema (108 nits), view_transform: ACES 1.1 HDR-cinema (108 nits & P3 lim), display_colorspace: ST2084-P3-D65}
    - !<Views> [Un-tone-mapped, Log, Raw]
  Rec.2100-PQ:
    - !<View> {name: ACES 1.1 HDR-video (1000 nits), view_transform: ACES 1.1 HDR-video (1000 nits & Rec.2020 lim), display_colorspace: Rec.2100-PQ}
    - !<View> {name: ACES 1.1 HDR-video (2000 nits), view_transform: ACES 1.1 HDR-video (2000 nits & Rec.2020 lim), display_colorspace: Rec.2100-PQ}
    - !<View> {name: ACES 1.1 HDR-video (4000 nits), view_transform: ACES 1.1 HDR-video (4000 nits & Rec.2020 lim), display_colorspace: Rec.2100-PQ}
    - !<Views> [Un-tone-mapped, Log, Raw]
  Rec.2100-HLG:
    - !<View> {name: ACES 1.1 HDR-video (1000 nits), view_transform: ACES 1.1 HDR-video (1000 nits & Rec.2020 lim), display_colorspace: Rec.2100-HLG}
    - !<Views> [Un-tone-mapped, Log, Raw]
```

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (KelSolaar) [reply opencolorio_config_aces/config/reference/generate/config.py:439]: Alternatively: `CSC - CLog3 CGamut`?

**Comment by KelSolaar** [comment]
Hi,

Been busy two weeks and I haven't got time to do much beside poking at the ordering and trying a few things.

I also removed the generated configs from the thread here as it clutters and slows down the discussions, they are available as a Gist.

Reference Config: https://gist.github.com/KelSolaar/1e2b054080e074e7f3529439470ee643/cd823328d060727ceaefb556120ec4709e8734d5

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (doug-walker) [reply opencolorio_config_aces/config/reference/generate/config.py:441]: Let's discuss at the meeting.  I'm fine with your example but the config in the Gist also has "CSC - ACES2065-1", which I don't like so much.  Basically I feel like there should be some distinction in naming between the official ACES spaces and the vendor spaces.

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (doug-walker) [reply opencolorio_config_aces/config/reference/generate/config.py]: Thanks for changing the name of that view.  Any disagreement on my proposed Description or setting it as the default view transform?

File: opencolorio_config_aces/config/reference/generate/config.py
Comment (KelSolaar) [reply opencolorio_config_aces/config/reference/generate/config.py:441]: `aces-dev` has them part of the ACES family:

image
image

and so do the `ACEStransformID`s:

- // <ACEStransformID>urn:ampas:aces:transformId:v1.5:ACEScsc.Academy.ACES_to_ACEScc.a1.0.3</ACEStransformID>
- // <ACEStransformID>urn:ampas:aces:transformId:v1.5:ACEScsc.Academy.ACES_to_LogC_EI800_AWG.a1.1.0</ACEStransformID>

Whether it is an oversight or not, I cannot comment! I think that we should strive to be as close to metal for the _Reference - Config_ and take some well documented liberties for the _Studio - Config_.

**Comment by KelSolaar** [comment]
Hello,

I have just implemented support for the `Look` transform!

Reference Config: https://gist.github.com/KelSolaar/1e2b054080e074e7f3529439470ee643/bb6cec792cc80ef4008ec05dc2a95feb45533dfb

**Comment by KelSolaar** [comment]
And with that last stack, we are done! 🎊 

- Support for user-friendly CSC names, not great for the Reference Config but... let's talk about it AGAIN tomorrow :)
- Support for encoding and categories.
- Support for default view transform

Reference Config: https://gist.github.com/KelSolaar/1e2b054080e074e7f3529439470ee643/317c6cca7577c069f75a64f5083961cf2cdb2643

**Review by michdolan — APPROVED** [review]
Looks great. Thanks @KelSolaar !

**Comment by michdolan** [comment]
Before we merge this, we should try to resolve the DSO issue. It's saying:

`Commit sha: 86f4d52, Author: Thomas Mansencal, Committer: Thomas Mansencal; The sign-off is missing.`

Since you have force pushed, it's possible that commit is no longer present, in which case this may just be a false error at this point. I wanted to check though.

**Comment by KelSolaar** [comment]
Yep I see that, will have to rebase everything!

**Comment by KelSolaar** [comment]
@michdolan : Should be good now!

**Comment by michdolan** [comment]
@doug-walker we can merge this once you approve. Merge is currently blocked by your requested changes. Thanks!

**Review by michdolan — APPROVED** [review]

File: opencolorio_config_aces/config/generation/common.py
Comment (doug-walker) [inline opencolorio_config_aces/config/generation/common.py:266]: In the Gist, the process_space for BlueLightFix is being set to scene_linear, but that role is ACEScg in this config.  The process_space must be set to ACES2065-1 for this LMT.  And, in fact, anything that is called an LMT must use that process space, by definition.

**Review by doug-walker — CHANGES_REQUESTED** [review]
Thanks for the additional work Thomas!  Please address the process_space issue before we merge.  

Ideally, it would be nice to have a Description added to the Un-tone-mapped view transform (either what I proposed, or something else).  Similarly, I think it would help having the top level config Description clarify that this is not intended as a replacement for the 1.2 ACES config (which some people have already gotten confused about on Slack).  For example: "This is an implementation of the aces-dev repo on GitHub using OCIO v2. It is a technical reference for developers and is not intended as a replacement for the previous general purpose ACES configs."   The Description stuff could be done in a follow-up, but I think it's important to correct the process_space issue now.

File: opencolorio_config_aces/config/generation/common.py
Comment (KelSolaar) [reply opencolorio_config_aces/config/generation/common.py:266]: Ah, good catch!

**Comment by KelSolaar** [comment]
@doug-walker : Done!

Reference Config: https://gist.github.com/KelSolaar/1e2b054080e074e7f3529439470ee643/bb51d19155be8867a072bf6fb3ca488a3eccc5df

**Review by doug-walker — APPROVED** [review]
Thanks Thomas!

**Pull request merged**
